app/controlplane/internal/biz/attestationstate.go

```diff
@@ -0,0 +1,236 @@
+	run, err := uc.checkWorkflowRunInWorkflow(ctx, workflowID, runID)
+	if err != nil {
+		return fmt.Errorf("failed to check workflow run: %w", err)
+	}
+
+	rawState, err := proto.Marshal(state)
+	if err != nil {
+		return fmt.Errorf("failed to marshal attestation state: %w", err)
+	}
+
+	encryptedState, err := encrypt(rawState, passphrase)
+	if err != nil {
+		return fmt.Errorf("failed to encrypt attestation state: %w", err)
+	}
+
+	if err := uc.repo.Save(ctx, run.ID, encryptedState); err != nil {
+		return fmt.Errorf("failed to save attestation state: %w", err)
+	}
+
+	return nil
+}
+
+func (uc *AttestationStateUseCase) Read(ctx context.Context, workflowID, runID, passphrase string) (*AttestationState, error) {
+	run, err := uc.checkWorkflowRunInWorkflow(ctx, workflowID, runID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to check workflow run: %w", err)
+	}
+
+	res, err := uc.repo.Read(ctx, run.ID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read attestation state: %w", err)
+	}
+
+	decryptedState, err := decrypt(res, passphrase)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decrypt attestation state: %w", err)
+	}
+
+	state := &v1.CraftingState{}
+	if err := proto.Unmarshal(decryptedState, state); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal attestation state: %w", err)
+	}
+
+	return &AttestationState{State: state}, nil
+}
+
+func (uc *AttestationStateUseCase) Reset(ctx context.Context, workflowID, runID string) error {
+	run, err := uc.checkWorkflowRunInWorkflow(ctx, workflowID, runID)
+	if err != nil {
+		return fmt.Errorf("failed to check workflow run: %w", err)
+	}
+
+	if err := uc.repo.Reset(ctx, run.ID); err != nil {
+		return fmt.Errorf("failed to reset attestation state: %w", err)
+	}
+
+	return nil
+}
+
+// checkWorkflowRunInWorkflow checks if the workflow run belongs to the provided workflow
+// This is important because the workflow is something that comes embedded in the auth token
+// so it can be used to make sure the user is not spoofing a different run that doesn't have access to
+func (uc *AttestationStateUseCase) checkWorkflowRunInWorkflow(ctx context.Context, workflowID, runID string) (*WorkflowRun, error) {
+	workflowUUID, err := uuid.Parse(workflowID)
+	if err != nil {
+		return nil, NewErrInvalidUUID(err)
+	}
+
+	runUUID, err := uuid.Parse(runID)
+	if err != nil {
+		return nil, NewErrInvalidUUID(err)
+	}
+
+	run, err := uc.wfRunRepo.FindByID(ctx, runUUID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to workflow run: %w", err)
+	} else if run == nil {
+		return nil, NewErrNotFound("workflow run")
+	}
+
+	if run.Workflow.ID != workflowUUID {
+		return nil, NewErrNotFound("workflow run")
+	}
+
+	return run, nil
+}
+
+// The following code is in charge of symmetric encryption and decryption of the attestation state
+// The only purpose is to have encryption at rest in the database
+const (
+	saltSize   = 16
+	iterations = 10000
```

Comment:
Why 10k? https://en.wikipedia.org/wiki/PBKDF2

I would also prefer to have the encryption/decryption logic in a separate module, if possible, so it can be reused later.